docs/source/app-dev/explicit-contract-disclosure.rst

```diff
@@ -0,0 +1,94 @@
+.. Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _explicit-contract-disclosure:
+
+(Experimental) Explicit contract disclosure
+###########################################
+
+Starting with Canton 2.6, explicit contract disclosure is introduced as an (experimental) feature
+that allows delegating contract read rights to non-stakeholder parties in an optimal manner.
+Until now, read delegation was possible as well through the following mechanisms:
```

Comment:
The way this is written a reader sort of needs to understand these two features, which we don't want them to understand at all, as they are sub-optimal.

I'd suggest to turn this around as follows: introduce a good example for off-ledger disclosure (a good one is IMHO reference data) and then explain how off-ledger disclosure is used to solve that. After explaining the desired solution, explain the existing alternatives and why they are not recommended.

Here are some notes form a recent conversation I've had about reference data for the exchange to serve a price-feed:

```
template PriceDatum with
    publisher : Party
    symbol : Text
    price : Decimal
    asOf : Time
  where
    signatory publisher

    nonconsuming choice PriceDatum_Read : PriceDatum
      with reader : Party
      controller reader
      do return this
```

and explain the topology where the exchange is connected to the domain on which their customers want to use the price feed.

If the exchange wants to bill customers based on the amount of reads they do then they can use:
```
template PriceDatum with
    publisher : Party
    symbol : Text
    price : Decimal
    asOf : Time
  where
    signatory publisher

    nonconsuming choice PriceDatum_Read : PriceDatum
      with reader : Party
      controller reader
      do return this
```

and use an off-ledger flow to bill the customer based on the number of validated tx nodes for `PriceDatum_Read`.

(this example requires further working out -- submitting now for consideration, as I need to head off)